### PR TITLE
Docs/778 menu flyout updates

### PR DESCRIPTION
**Pull request by Rykus0**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

closes #778 

## What is the new behavior?

Updates docs and examples for updated API and hooks

### CODE REVIEW

File: website/docs/development/headless-styles/Menu.mdx [hunk 1]
```diff
@@ -57,14 +57,15 @@ For a single level menu, just call the helper with no options passed in.
 
 ### Submenus
 
-When you need a submenu, set `kind` to `submenu`, add your `label` options, and set `isSubmenuExpanded` when it is displayed.
+When you need a submenu, set `kind` to `submenu` and add your `label` options.
+When a submenu is displayed, the `aria-expanded` on the triggering element should be set to `"true"`.
```
Comment (caseybaggz) [inline website/docs/development/headless-styles/Menu.mdx:61]: Not seeing a reason why this line is necessary.

```suggestion
```

File: website/docs/development/react-utils/use-submenu-interaction.mdx [hunk 2]
```diff
@@ -57,47 +57,27 @@ interface SubmenuInteractionProps {
...
-    </li>
-  )
-}
-
 export function Submenu(props) {
   const submenuStyles = getMenuProps({
     kind: 'submenu',
     label: props.label,
   })
   // highlight-next-line
```
Comment (caseybaggz) [inline website/docs/development/react-utils/use-submenu-interaction.mdx:65]: Let's kill this and instead use the binding to highlight all the lines for the submenu stuff in the code:
https://docusaurus.io/docs/markdown-features/code-blocks#highlighting-with-metadata-string

File: website/docs/development/headless-styles/Menu.mdx [hunk 3]
```diff
@@ -94,26 +84,14 @@ Hover should be avoided since it does not convey intent and is difficult for som
...
-[roving tabindex](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets#technique_1_roving_tabindex)
-for managing focus and tab order.
-
-:::note
-
-If the menu is not shown until another element is interacted with (e.g., pressing a button),
-then it is considered a [fly-out](https://www.w3.org/WAI/tutorials/menus/flyout/) or pop-up menu.
-
-:::
+To ensure Menus are keyboard accessible, we recommend using our Menu Hooks in the React Utils library. For full support, see [useMenuInteraction](../react-utils/use-menu-interaction.mdx) for base-level popup menus, [useSubmenuInteraction](../react-utils/use-submenu-interaction.mdx) for nested submenus, and [useRovingTabIndex] for menu item focus management.
```
Comment (caseybaggz) [inline website/docs/development/headless-styles/Menu.mdx:87]: Missing the link on the Roving index

File: website/src/components/Menu/SubmenuExample.preview.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import CodeBlock from '@theme/CodeBlock'
 export function SubmenuExamplePreview() {
```
Comment (caseybaggz) [inline website/src/components/Menu/SubmenuExample.preview.js:4]: I'm not sure why we added all the highlights in the example code blocks? Let's remove those to stay consistent.

File: website/docs/development/headless-styles/Menu.mdx (hunk 3 above)
Comment (Rykus0) [reply website/docs/development/headless-styles/Menu.mdx:87]: oof, thank you!

File: website/src/components/Menu/SubmenuExample.preview.js (hunk 4 above)
Comment (Rykus0) [reply website/src/components/Menu/SubmenuExample.preview.js:4]: You had asked me to add highlights, but maybe I misinterpreted it.  I can definitely remove them.

Comment (Rykus0) [reply website/src/components/Menu/SubmenuExample.preview.js:4]: Yep, I added them to the wrong spot

**Review by caseybaggz — APPROVED** [review]

**Pull request merged**